src/borg/archiver.py

```diff
@@ -1474,12 +1474,10 @@ def do_prune(self, args, repository, manifest, key):
                              '"keep-secondly", "keep-minutely", "keep-hourly", "keep-daily", '
                              '"keep-weekly", "keep-monthly" or "keep-yearly" settings must be specified.')
             return self.exit_code
-        archives_checkpoints = manifest.archives.list(sort_by=['ts'], reverse=True)  # just a ArchiveInfo list
         if args.prefix:
             args.glob_archives = args.prefix + '*'
-        if args.glob_archives:
-            regex = re.compile(shellpattern.translate(args.glob_archives))
-            archives_checkpoints = [arch for arch in archives_checkpoints if regex.match(arch.name) is not None]
+        archives_checkpoints = manifest.archives.list(glob=args.glob_archives, match_end=False,
+                                                      sort_by=['ts'], reverse=True)
```

Comment:
I think it'd be better to write this more along the lines of `if name.endswith('.checkpoint'): regex.match(name[:-len('.checkpoint')]) else: regex.match(name)`, because this way it effectively means `...*`, which includes more than just checkpoints.